http.go

```diff
@@ -56,8 +58,55 @@ func matchRegularExpressions(reader io.Reader, config HTTPProbe) bool {
 
 func probeHTTP(target string, w http.ResponseWriter, module Module) (success bool) {
 	var isSSL, redirects int
+	var dialProtocol, fallbackProtocol string
+
 	config := module.HTTP
 
+	if module.HTTP.Protocol == "" {
+		module.HTTP.Protocol = "tcp"
+	}
+
+	if module.HTTP.Protocol == "tcp" && module.HTTP.PreferredIpProtocol == "" {
+		module.HTTP.PreferredIpProtocol = "ip6"
+	}
+	if module.HTTP.PreferredIpProtocol == "ip6" {
+		fallbackProtocol = "ip4"
+	} else {
+		fallbackProtocol = "ip6"
+	}
+
+	dialProtocol = module.HTTP.Protocol
+	if module.HTTP.Protocol == "tcp" {
+		target_url, err := url.Parse(target)
+		if err != nil {
+			return false
+		}
+		target_host, _, err := net.SplitHostPort(target_url.Host)
+		// If split fails, assuming it's a hostname without port part
+		if err != nil {
+			target_host = target_url.Host
+		}
+		ip, err := net.ResolveIPAddr(module.HTTP.PreferredIpProtocol, target_host)
```

Comment:
I was reading the code and came across this and tracked it to this PR:
This causes another DNS roundtrip for probe, correct? Or do I miss something?